# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [5]:
import pandas  as pd
import numpy as np
import random 

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [6]:
randomlist_1_4 = []
for i in range(0,100):
    n = random.randint(1,4)
    randomlist_1_4.append(n)
df_1_4=pd.Series(randomlist_1_4)
df_1_4
# alternate
# ds1=pd.Series(np.random.randint(1,4,100))
# ds1

0     3
1     1
2     3
3     1
4     3
     ..
95    4
96    4
97    2
98    1
99    1
Length: 100, dtype: int64

In [8]:
random_list_1_3 = []
for i in range(0,100):
    n = random.randint(1,3)
    random_list_1_3.append(n)
df_1_3=pd.Series(random_list_1_3)
df_1_3

0     2
1     2
2     3
3     3
4     2
     ..
95    3
96    2
97    1
98    3
99    3
Length: 100, dtype: int64

In [10]:
df_10K_30K=pd.Series(np.random.randint(10000,30000,100))
df_10K_30K

0     25928
1     28811
2     11110
3     25728
4     15049
      ...  
95    23938
96    23274
97    16027
98    10843
99    29479
Length: 100, dtype: int32

### Step 3. Let's create a DataFrame by joinning the Series by column

In [11]:
df_random=pd.DataFrame({
    "random_1-4":df_1_4,
    "randdom_1-3":df_1_3,
    "random_10K-30K":df_10K_30K
})
df_random

random_1-4  randdom_1-3  random_10K-30K
0            3            2           25928
1            1            2           28811
2            3            3           11110
3            1            3           25728
4            3            2           15049
..         ...          ...             ...
95           4            3           23938
96           4            2           23274
97           2            1           16027
98           1            3           10843
99           1            3           29479

[100 rows x 3 columns]

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [12]:
df_random.columns=['bedrs','bathrs','price_sqr_meter']
df_random
# OR
# df_random.rename(columns = {'random_1-4':'bedrs','random_1-3':'bathrs','random_10K-30K':'price_sqr_meter'}, inplace = True)

bedrs  bathrs  price_sqr_meter
0       3       2            25928
1       1       2            28811
2       3       3            11110
3       1       3            25728
4       3       2            15049
..    ...     ...              ...
95      4       3            23938
96      4       2            23274
97      2       1            16027
98      1       3            10843
99      1       3            29479

[100 rows x 3 columns]

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [17]:
bigcolumn = pd.concat(objs=[df_1_4,df_1_3,df_10K_30K],ignore_index=True)
bigcolumn.name='bigcolumn'
bigcolumn

0          3
1          1
2          3
3          1
4          3
       ...  
295    23938
296    23274
297    16027
298    10843
299    29479
Name: bigcolumn, Length: 300, dtype: int64

### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [19]:
df_99=pd.DataFrame(pd.concat([df_1_4,df_1_3,df_10K_30K]))
df_99

0
0       3
1       1
2       3
3       1
4       3
..    ...
95  23938
96  23274
97  16027
98  10843
99  29479

[300 rows x 1 columns]

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [20]:
df_99.reset_index(drop=True)

0
0        3
1        1
2        3
3        1
4        3
..     ...
295  23938
296  23274
297  16027
298  10843
299  29479

[300 rows x 1 columns]